File: Problems/Algorithms/589. N-Ary Tree Preorder Traversal/n_ary_preorder.go
package main

// Definition for a Node.
type Node struct {
	Val      int
	Children []*Node
}

func preorder(root *Node) []int {
	if root == nil {
		return []int{}
	}

	if root.Children == nil {
		return []int{root.Val}
	}

	ans := []int{root.Val}
	for _, child := range root.Children {
		ans = append(ans, preorder(child)...)
	}
	return ans
}

File: Problems/Algorithms/589. N-Ary Tree Preorder Traversal/n_ary_preorder.py
#!/usr/bin/env python3

from typing import List


# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        if root is None:
            return []

        if root.children is None:
            return [root.val]

        ans = [root.val]
        for child in root.children:
            ans += self.preorder(child)
        return ans
